### PR TITLE
Added proto files for comms Envelope and use in OMS/IMS

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

This is the first iteration of using protobuf in comms (ref #890  .

- Added build.rs to build proto files (using prost)
- Use protobuf envelope in IMS and OMS

Still a number of over-the-wire messages to convert.

## Motivation and Context

Progress on #890

## How Has This Been Tested?

New unit tests + existing tests updated

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: comms/src/message/message_envelope.rs [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    fn try_from(frames: FrameSet) -> Result<Self, Self::Error> {
+        if frames.len() != FRAMES_PER_MESSAGE {
+            return Err(MessageError::InvalidMultipartMessageLength);
+        }
+
+        Ok(MessageEnvelope { frames })
+    }
+}
+
+//#[cfg(test)]
```
Comment (neonknight64) [inline comms/src/message/message_envelope.rs]: Can these tests be added back?

File: comms/src/message/inbound_message.rs [hunk 2]
```diff
@@ -20,34 +20,25 @@
...
-    /// The message envelope header
+    /// The deserialized message envelope header
     pub envelope_header: MessageEnvelopeHeader,
     /// The connected peer which sent this message
     pub source_peer: Peer,
-    /// The version of the incoming message envelope
-    pub version: u8,
     /// The raw message envelope
-    pub body: Frame,
+    pub body: Vec<u8>,
```
Comment (neonknight64) [inline comms/src/message/inbound_message.rs]: In MessageEnvelope Frames and Framesets are still used, shouldn't this also be a Frame.

Comment (sdbondi) [reply comms/src/message/inbound_message.rs]: Frame/FrameSet were indented to represent Frames coming from zmq connections. Once a frame has been deserialized, any "byte blob" field is not a Frame.

File: comms/src/message/message_envelope.rs (hunk 1 above)
Comment (sdbondi) [reply comms/src/message/message_envelope.rs]: I've added tests for the impls on the new message types. Some of these tests won't apply, specifically MessageEnvelope won't be needed.

Comment (sdbondi) [reply comms/src/message/message_envelope.rs]: I'll add back the applicable tests in a subsequent PR

**Pull request merged**